decidim-elections/app/permissions/decidim/elections/trustee_zone/permissions.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Elections
+    module TrusteeZone
+      class Permissions < Decidim::DefaultPermissions
+        def permissions
+          return permission_action unless permission_action.scope == :trustee_zone
+
+          case permission_action.subject
+          when :trustee
+            toggle_allow(trustee?) if permission_action.action == :view
+          when :user
+            allow! if permission_action.action == :update_profile
```

Comment:
@mrcasals I had to support this action because the TrusteeZone controller is using the UserProfile concern, that includes [this](https://github.com/decidim/decidim/blob/develop/decidim-core/app/controllers/concerns/decidim/user_profile.rb#L23).

First, I though about removing that `before_action` block from the concern, as no permissions class is referring to that `update_profile` action. But then I noticed that some of the user profile pages controllers are not enforcing permissions at all (for example, [newsletters](https://github.com/decidim/decidim/blob/develop/decidim-core/app/controllers/decidim/newsletters_controller.rb#L22)), so I guess that this could result on a security issue.

What do you think as the creator of the permissions system? Am I right with this? Should I keep this or can I safely remove the `before_action` block?

Reply:
Uf, I'm not sure about removing the `before_action` block. In this case, I'd leave the code as it is and add a note in this permission class explaining why this action is needed.

At some point I think we should rethink how to approach permissions, because this is becoming a mess and it's super hard to reason about 😞